### PR TITLE
added JWT HW requirements

### PR DESCRIPTION
**Pull request by boroda4436**

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -0,0 +1,8 @@
+db.driver=com.mysql.cj.jdbc.Driver
+db.url=jdbc:mysql://localhost:3306/ticket_app?serverTimeZone=UTC
+db.username=root
+db.password=12345678
+
+hibernate.show_sql=true
+hibernate.hbm2ddl.auto=create-drop
+hibernate.dialect=org.hibernate.dialect.MySQL5Dialect
```
Comment (Resci) [inline src/main/resources/application.properties:8]: Чому не 8?

**Review by Resci — APPROVED** [review]

File: src/main/java/mate/academy/dao/impl/RoleDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import mate.academy.exception.DataProcessingException;
+import mate.academy.model.Role;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class RoleDaoImpl extends AbstractDao<Role, Long> implements RoleDao {
+
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/dao/impl/RoleDaoImpl.java:15]:
```suggestion
```

File: src/main/java/mate/academy/model/Role.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "roles")
+public class Role {
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/model/Role.java:13]: can we use Lombok?

File: src/main/java/mate/academy/security/CustomUserDetailsService.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        Optional<User> userOptional = userService.findByEmail(email);
+
+        UserBuilder builder;
+        if (userOptional.isPresent()) {
+            builder = org.springframework.security.core.userdetails.User.withUsername(email);
+            builder.password(userOptional.get().getPassword());
+            builder.roles(userOptional.get().getRoles()
+                    .stream()
+                    .map(x -> x.getRoleName().name())
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/security/CustomUserDetailsService.java:30]:
```suggestion
                    .map(r -> r.getRoleName().name())
```

File: src/main/java/mate/academy/service/mapper/ProductDtoMapper.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
+package mate.academy.service.mapper;
+
+import mate.academy.model.Product;
+import mate.academy.model.dto.ProductRequestDto;
+import mate.academy.model.dto.ProductResponseDto;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ProductDtoMapper {
+
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/service/mapper/ProductDtoMapper.java:10]:
```suggestion
```

File: src/main/java/mate/academy/validation/PasswordValidator.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        Object fieldValue = new BeanWrapperImpl(registrationDto)
+                .getPropertyValue(field);
+        Object fieldMatchValue = new BeanWrapperImpl(registrationDto)
+                .getPropertyValue(fieldMatch);
+
+        if (fieldValue != null) {
+            return fieldValue.equals(fieldMatchValue);
+        } else {
+            return false;
+        }
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/validation/PasswordValidator.java:29]:
```suggestion
        return fieldValue != null && fieldValue.equals(fieldMatchValue);
```

File: README.md [hunk 7]
```diff
@@ -1 +1,11 @@
-# jv-homework-template
+# Spring Security. Stateless
+
+1. Implement method `protected void configure(HttpSecurity http)` in the `SecurityConfig` class
+1. Permit access for all user to "/register", "/login", "/inject" endpoints
+1. Permit access only for ADMIN to all endpoints marked with @DeleteMapping annotation
```
Comment (KseniiaMakarova) [inline README.md:5]:
```suggestion
1. Permit access only for ADMIN to all endpoints marked with `@DeleteMapping` annotation
```

File: src/main/java/mate/academy/config/AppConfig.java [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+import org.springframework.orm.hibernate5.LocalSessionFactoryBean;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@Configuration
+@PropertySource("classpath:application.properties")
+@ComponentScan(basePackages = "mate.academy")
+public class AppConfig {
+    @Autowired
+    private Environment environment;
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/config/AppConfig.java:21]: can we use autowiring through constructor?

File: src/main/java/mate/academy/controller/AuthenticationController.java [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class AuthenticationController {
+    private final AuthenticationService authenticationService;
+    private final UserMapper userMapper;
+
+    public AuthenticationController(AuthenticationService authenticationService,
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/controller/AuthenticationController.java:18]:
```suggestion
    @Autowired
    public AuthenticationController(AuthenticationService authenticationService,
```
for consistency, we should better use annotation everywhere

**Review by KseniiaMakarova — APPROVED** [review]

File: src/main/java/mate/academy/model/Role.java (hunk 3 above)
Comment (boroda4436) [reply src/main/java/mate/academy/model/Role.java:13]: no. I haven't show him Lombok in videos

File: src/main/java/mate/academy/controller/AuthenticationController.java (hunk 9 above)
Comment (boroda4436) [reply src/main/java/mate/academy/controller/AuthenticationController.java:18]: not sure this is required. I haven't used @Autowired over constructors

File: src/main/resources/application.properties (hunk 1 above)
Comment (boroda4436) [reply src/main/resources/application.properties:8]: basically, it doesn't matter. Both could be used.

**Pull request merged**